Doctrine/Annotation/AnnotationReader.php

```diff
@@ -31,36 +33,105 @@ public function __construct(Reader $reader)
         $this->reader = $reader;
     }
 
-    /**
+     /**
      * reads the entity and returns a set of annotations
      *
      * @param object $entity
      * @param string $type
+     * @param array $fields
      *
      * @return Annotation[]
      */
-    private function getPropertiesByType($entity, $type)
+    private function getPropertiesByType($entity, $type, $fields = array())
     {
         $properties = $this->readClassProperties($entity);
 
-        $fields = array();
         foreach ($properties as $property) {
-            $annotation = $this->reader->getPropertyAnnotation($property, $type);
 
+            $property->setAccessible(true); 
+            $annotation = $this->reader->getPropertyAnnotation($property, $type);
+            
             if (null === $annotation) {
                 continue;
             }
 
-            $property->setAccessible(true);
-            $annotation->value = $property->getValue($entity);
-            $annotation->name = $property->getName();
+            if($type == $this::FIELDS_CLASS) {
+                
+                $fields = $this->processFieldsAnnotation($property, $annotation, $entity, $fields);
+            }
+            else {
```

Comment:
remove new line

Reply:
Dear Florian,

Thank you for you're time reviewing my PR. I have added the following changes (see commits):

Added fix for inserting array of objects instead of doctrine annotation data.
Added test testReadPropertiesMultipleFields in AnnotationReaderTest.php
Added extra tests for exceptions:
testGetFields_NoParentGetter
testGetFields_UnkownParentGetterMethod
testGetFields_NoFieldAlias
testGetFields_NoFieldGetter
testGetFields_UnknownFieldGetter

Removed newlines and added missing spaces.
Can you please reevaluate my PR? If you require any further information, feel free to contact me.

Kind regards,

Jack Kwakman